src/main/java/com/puppycrawl/tools/checkstyle/filters/PatchXpathFilterElement.java

```diff
@@ -245,12 +272,18 @@ private DetailAST getEventAst(TreeWalkerAuditEvent event) {
             }
             curNode = toVisit;
         }
-        return childAstLineNoSet;
+        return childAstLineNoMap;
     }
 
-    private void addChildAstLineNo(Set<Integer> childAstLineNoSet, DetailAST ast) {
+    private void setChildAstLineNo(Map<String, Integer> childAstLineNoMap, DetailAST ast) {
         if (ast != null) {
-            childAstLineNoSet.add(ast.getLineNo());
+            final int lineNo = ast.getLineNo();
+            if (lineNo <= childAstLineNoMap.get(MIN)) {
+                childAstLineNoMap.put(MIN, lineNo);
+            }
+            else if (lineNo >= childAstLineNoMap.get(MAX)) {
+                childAstLineNoMap.put(MAX, lineNo);
+            }
```

Comment:
please use strict operators `<` and `>`. We don't need to rewrite the same values.

Reply:
done